### PR TITLE
Adds device-challenge-poll transformer for custom URI

### PR DESCRIPTION
**Pull request by shawyu-okta**
## PR Checklist

- [ ] Have you verified the basic functionality for this change?
- [ ] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [ ] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [ ] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-548962](https://oktainc.atlassian.net/browse/OKTA-548962)

### CODE REVIEW

File: src/v3/src/transformer/oktaVerify/transformOktaVerifyChallengePoll.ts [hunk 1]
```diff
@@ -35,82 +37,88 @@ export const transformOktaVerifyChallengePoll: IdxStepTransformer = ({ transacti
...
-  const autoChallenge = getUIElementWithName('autoChallenge', uischema.elements) as FieldElement;
-  data.autoChallenge = autoChallenge?.options?.inputMeta?.value;
+  if (selectedMethod.type === 'push') {
+    // Need to initialize autoChallenge checkbox if it is set otherwise it will not display in UI
+    const autoChallenge = getUIElementWithName('autoChallenge', uischema.elements) as FieldElement;
+    data.autoChallenge = autoChallenge?.options?.inputMeta?.value;
 
-  // @ts-ignore OKTA-496373 correctAnswer is missing from interface
-  const correctAnswer = relatesTo?.value?.contextualData?.correctAnswer;
-  if (selectedMethod.type === 'push' && correctAnswer) {
```
Comment (shawyu-okta) [inline src/v3/src/transformer/oktaVerify/transformOktaVerifyChallengePoll.ts:44]: Note: I broke this logic apart. It used to be like this:
```js
if (condition1 && condition2) {
  // ... do if c1 and c2 are true
} else if (condition1) {
  // ... do if only c1 is true
}
```
now it is
```js
if (condition1) {
  if (condition2) {
    // ... do if c1 and c2 are true
  } else {
    // ... do if only c1 is true
  }
}
```

File: src/v3/src/transformer/i18n/transformOpenOktaVerifyButton.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/*
```
Comment (shawyu-okta) [inline src/v3/src/transformer/i18n/transformOpenOktaVerifyButton.ts:1]: Note to self: Add transformer unit test before merge.

File: src/v3/src/components/AuthCoin/AuthCoinConfig.ts [hunk 3]
```diff
@@ -154,6 +154,13 @@ const AuthCoinByAuthenticatorKey: Record<string, AuthCoinConfig> = {
     description: loc('factor.totpHard.yubikey', 'login'),
     iconClassName: 'mfa-yubikey',
   },
+  [CHALLENGE_METHOD.CUSTOM_URI]: {
+    icon: OktaVerifyIcon,
+    name: 'mfa-okta-verify',
+    customizable: false,
+    description: loc('factor.totpSoft.oktaVerify', 'login'),
+    iconClassName: 'mfa-okta-verify',
+  },
```
Comment (glenfannin-okta) [inline src/v3/src/components/AuthCoin/AuthCoinConfig.ts:163]: nit: Can you move this up under `[AUTHENTICATOR_KEY.CUSTOM_OTP]: { ... }` since it looks like these objects are ordered alphabetically by key?

File: src/v3/src/components/OpenOktaVerifyButton/OpenOktaVerifyButton.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    src={src}
+    style={{
+      display: 'none',
+    }}
+  />
+);
+
+const OpenOktaVerifyButton: UISchemaElementComponent<{
+  uischema: OpenOktaVerifyButtonElement
+}> = ({ uischema }) => {
```
Comment (glenfannin-okta) [inline src/v3/src/components/OpenOktaVerifyButton/OpenOktaVerifyButton.tsx:41]: Just a question, does having `FP` or `FastPass` in the component/Element name help or would that be too wordy/redundant? It may be redundant since this is still technically OV, just a slight variation.

**Review by glenfannin-okta — APPROVED** [review]
Lgtm. 

Quick question, do you think this would be a good candidate to add to `test/e2e/tests/a11y.ts` to test the view against a11y checks and capture a screenshot?

File: src/v3/src/components/AuthCoin/AuthCoinConfig.ts (hunk 3 above)
Comment (shawyu-okta) [reply src/v3/src/components/AuthCoin/AuthCoinConfig.ts:163]: I was trying to decide if I should group these by the first part of the constant name (e.g. AUTHENTICATOR_KEY, CHALLENGE_METHOD) and then alphabetical within those groups, or just alphabetical across all. What do you think?

File: src/v3/src/components/OpenOktaVerifyButton/OpenOktaVerifyButton.tsx (hunk 4 above)
Comment (shawyu-okta) [reply src/v3/src/components/OpenOktaVerifyButton/OpenOktaVerifyButton.tsx:41]: I think adding FP might be helpful context in the name without lengthening it too much 👍

File: src/v3/src/components/AuthCoin/AuthCoinConfig.ts (hunk 3 above)
Comment (glenfannin-okta) [reply src/v3/src/components/AuthCoin/AuthCoinConfig.ts:163]: Ohhh, sorry, my eyes are playing tricks on me, I think here is fine 😅 since this is ordered by the first part of the constant. Sorry about that my brain totally ignored the `CHALLENGE_METHOD` part 😅

**Comment by shawyu-okta** [comment]
> Quick question, do you think this would be a good candidate to add to test/e2e/tests/a11y.ts to test the view against a11y checks and capture a screenshot?

Good callout - I'll add this along with the unit tests I owe
